src/carma-platform/guidance_scripts.py

```diff
@@ -163,5 +179,461 @@ def run_crosstrack_analysis(
 
     return (is_passed, stats, plt.gcf(), cross_tracks, timestamps)
 
+def process_actual_path(odom_data):
+    """Helper function that processes actual path data."""
+    return np.array([[point[0], point[1]] for point in odom_data])
+
+def process_planned_path(traj_plans):
+    """Helper function that processes planned path data with duplicate removal."""
+    planned_path = []
+    last_planned_point = None
+
+    for plan in traj_plans:
+        for point in plan:
+            if last_planned_point is None:
+                planned_path.append(point)
+                last_planned_point = point
+            else:
+                dist = np.linalg.norm(np.array(point) - np.array(last_planned_point))
+                if dist > 0.25:  # 0.25m threshold
+                    planned_path.append(point)
+                    last_planned_point = point
+                    break
```

Comment:
Wouldn't this break result in only 2 points being stored for each plan?

Reply:
Kind of yeah. We are actually just getting 1 point from each plan as we go. Because each plan in the traj_plans have 6 seconds worth of points. Many of the points would just repeat when we iterate of the plans, so it is taking the closest first point from each plan basically.